src/jit/codegenarmarch.cpp

```diff
@@ -781,14 +781,12 @@ void CodeGen::genCodeForArrOffset(GenTreeArrOffs* arrOffset)
         var_types elemType = arrOffset->gtArrElemType;
         unsigned  offset   = genOffsetOfMDArrayDimensionSize(elemType, rank, dim);
 
-// Load tmpReg with the dimension size and evaluate
-// tgtReg = offsetReg*dim_size + indexReg.
+        // Load tmpReg with the dimension size and evaluate
+        // tgtReg = offsetReg*tmpReg + indexReg.
+        emit->emitIns_R_R_I(ins_Load(TYP_INT), EA_PTRSIZE, tmpReg, arrReg, offset);
```

Comment:
@briansull I had assumed that "array offset" referred to a ptr diff.  i.e.  &a[95] - &a[0].  So I have assumed this is a pure math node no GC references needed to be reported or maintained.  However the EA_4BYTE looks suspicious especially in the arm64 code.